Makefile

```diff
@@ -0,0 +1,132 @@
+
+VIRTUALENV := $(shell poetry env info | grep Path: | sed 's/Path:[[:space:]]*//')
+PYTHONFILES := $(shell find website -name '*.py')
+MIGRATIONS := $(shell find website -name '*.py' | grep migrations)
+DOCSFILES := $(shell find docs -name '*.rst')
+
+PORT ?= 8000
+
+ifdef CI
+	POETRY_FLAGS := $(POETRY_FLAGS) --no-interaction
+	BLACK_FLAGS := $(BLACK_FLAGS) --quiet
+	DOCKER_FLAGS := $(DOCKER_FLAGS) --quiet
+endif
+
+help:
+	@echo "\033[1mUSAGE\033[0m"
+	@echo "  \033[4mmake\033[0m [options] <target>"
+	@echo
+	@echo "\033[1mOPTIONS\033[0m"
+	@echo "The following options can be set with \033[4mmake\033[0m \033[0;32mOPTION\033[0m=\033[0;34mvalue\033[0m"
+	@echo "  \033[0;32mPORT\033[0m=\033[0;34m$(PORT)\033[0m"
+	@echo "  \033[0;32mPOETRY_FLAGS\033[0m=\033[0;34m$(POETRY_FLAGS)\033[0m"
+	@echo "  \033[0;32mBLACK_FLAGS\033[0m=\033[0;34m$(BLACK_FLAGS)\033[0m"
+	@echo
+	@echo "\033[1mAVAILABLE TARGETS\033[0m"
+	@grep '^[[:alnum:]_-]*:.* ##' $(MAKEFILE_LIST) \
+	| sort | awk 'BEGIN {FS=":.* ## "}; {printf "  \033[0;36m%-25s\033[0m%s\n", $$1, $$2};'
+
+.make:
+	mkdir .make
+
+run: $(VIRTUALENV) website/db.sqlite3 ## Run a local webserver on PORT
+	poetry run website/manage.py runserver $(PORT)
+
+$(VIRTUALENV): poetry.lock pyproject.toml
+	poetry $(POETRY_FLAGS) install
+	@touch $(VIRTUALENV)
+
+deps: $(VIRTUALENV) ## Install all the required dependencies
+
+website/db.sqlite3: $(VIRTUALENV) $(MIGRATIONS)
+	poetry run website/manage.py migrate
+
+migrate: ## Run all database migrations
+	poetry run website/manage.py migrate
+
+makemigrations: ## Automatically create migration scripts
+	poetry run website/manage.py makemigrations
+
+createsuperuser: $(VIRTUALENV) website/db.sqlite3 ## Create a superuser for your local concrexit
+	poetry run website/manage.py createsuperuser
+
+createfixtures: $(VIRTUALENV) website/db.sqlite3 ## Create dummy database entries
+	poetry run website/manage.py createfixtures -a
+
+.make/fmt: .make $(VIRTUALENV) $(PYTHONFILES)
+	poetry run black $(BLACK_FLAGS) website
+	@touch .make/fmt
+
+fmt: .make/fmt ## Format python code with black
+
+blackcheck: $(VIRTUALENV) $(PYTHONFILES) ## Check if everything is formatted correctly
+	poetry run black $(BLACK_FLAGS) --check website
+
+.make/check: .make $(VIRTUALENV) $(PYTHONFILES)
+	poetry run python website/manage.py check
+	@touch .make/check
+
+check: .make/check ## Run internal Django tests
+
+.make/templatecheck: .make $(VIRTUALENV)
+	poetry run python website/manage.py templatecheck --project-only
+	@touch .make/templatecheck
+
+templatecheck: .make/templatecheck ## Test the templates
+
+.make/migrationcheck: .make $(VIRTUALENV) $(PYTHONFILES)
+	poetry run python website/manage.py makemigrations --no-input --check --dry-run
+	@touch .make/migrationcheck
+
+migrationcheck: .make/migrationcheck ## Check if migrations are created
+
+.coverage: $(VIRTUALENV) $(PYTHONFILES)
+	poetry run coverage run website/manage.py test website/
+
+tests: .coverage ## Run tests with coverage
```

Comment:
Wouldn't it make more sense to call this command 'coverage' and have it run the tests _and_ generate the coverage reports?

Reply:
I think you missed the `coverage` task which does exactly this. The tests command can be used if you don't care about the report

Reply:
Oh I missed that it also runs the tests themselves